internal/contour/snapshot.go

```diff
@@ -0,0 +1,55 @@
+package contour
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
+
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v2"
+)
+
+// SnapshotHandler implements xds snapshot cache
+type SnapshotHandler struct {
```

Comment:
Maybe a more evocative name would be `Snapshotter`, since it's not clear what this is a handler for or of. Probably `CacheHandler` is also not a great name any more for the same reason :)